libraries/AP_LandingGear/AP_LandingGear.cpp

```diff
@@ -208,6 +233,30 @@ void AP_LandingGear::update()
         
         gear_state_current = gear_state_new;
     }
+
+    /*
+      check for height based triggering
+     */
+    int16_t alt_m = constrain_int16(height_above_ground_m, 0, INT16_MAX);
+
+    if (hal.util->get_soft_armed()) {
+        // only do height based triggering when armed
+        if ((!_deployed || !_have_changed) &&
+            _deploy_alt > 0 &&
+            alt_m <= _deploy_alt &&
+            _last_height_above_ground > _deploy_alt) {
+            deploy();
+        }
+        if ((_deployed || !_have_changed) &&
+            _retract_alt > 0 &&
+            _retract_alt >= _deploy_alt &&
```

Comment:
I think it should just be bigger, not allow them to be set at same altitude.

Reply:
thats why it is an integer